src/ant.cpp

```diff
@@ -382,6 +383,30 @@ namespace full_coverage_path_planner
         gridNode_t new_location,
         bool add_to_path)
     {
+        int cx = current_location.pos.x;
+        int cy = current_location.pos.y;
+
+        int nx = new_location.pos.x;
+        int ny = new_location.pos.y;
+
+        if (cx+1 == nx)
+        {
+            direction = EAST;
+        }
+        else if (cx-1 == nx)
+        {
+            direction = WEST;
+        }
+        else if (cy+1 == ny)
+        {
+            direction = NORTH;
+        }
+        else
+        {
+            direction = SOUTH;
+        }
+
```

Comment:
This could use some work.

Any time the ant moves, the direction of movement is calculated from the changed grid index.

This should actually be created by the program when it initially generates the movement, not later on as is depicted currently.